iothub/device/src/ClientFactory.cs

```diff
@@ -80,27 +81,16 @@ public static InternalClient Create(string hostname, string gatewayHostname, IAu
 
             IotHubConnectionStringBuilder connectionStringBuilder = IotHubConnectionStringBuilder.Create(hostname, gatewayHostname, authenticationMethod);
 
+            // If the file upload transport settings hasn't been specified, we will create one using the client certificate on the connection string
+            EnsureOptionsIsSetup(connectionStringBuilder.Certificate, ref options);
```

Comment:
This step is valid only if the device auth mechanism is x509 certs, it should be inside this block: `if (authenticationMethod is DeviceAuthenticationWithX509Certificate)`

Reply:
we are going to need the options initialized regardless of the auth method.